db/migrations/20210129201213_single_name_field.sql

```diff
@@ -0,0 +1,15 @@
+-- migrate:up
+-- Add new single name field
+ALTER TABLE users ADD COLUMN name VARCHAR;
+COMMENT ON COLUMN users.name IS 'Name of user, may or may not be full name.';
+
+-- Migrate names over
+UPDATE users SET name = users.first_name || ' ' || users.last_name;
+
+ALTER TABLE users ALTER COLUMN name SET NOT NULL;
+
+-- Drop old name fields
+ALTER TABLE users DROP COLUMN first_name CASCADE, DROP COLUMN last_name CASCADE, DROP COLUMN preferred_name CASCADE;
+
+-- migrate:down
+
```

Comment:
Don't forget the `down` which reverts the DB to the way it was *before* this migration.

Reply:
Won't this run into the issue of creating two non-nullable fields without any values?

Reply:
It seems the "official" method of doing this is exactly like you've done in the up. Create it `NULL`-able and then set it to `NOT NULL` later. Since migrations happen in a transaction, you can't be left in an inconsistent state.